woocommerce/woocommerce-ios
# Blaze: Add loading and error states for target device and language picker views

Closes: #11655 


## Description

This PR refactors the `MultiSelectionView` to not include `NavigationView` and move the navigation setup to the picker views for Blaze target options. This is to make it flexible to add states to the views.

Changes:
- Injected selected items to the view model of the target option views instead of the views.
- Added states for the target option views.
- Disabled the Save button if the state is syncing/error or if the selected items are empty.
- Moved the navigation view to the target option views and removed it from the `MultiSelectionView`.

## Testing instructions

- Log in to a store with at least one existing product and is eligible for Blaze.
- Navigate to the Products tab and select any public product.
- On the product form, select Promote with Blaze.
- Notice that the campaign creation form is presented. Select the device field.
- Confirm the following:
  - Selection works properly. 
  - The device field on the creation form is updated following the selected items.
  - The Save button is enabled when there is data on the screen, and any item is selected.
  -  Hardcode the view's state - confirm that the UI is updated correctly.
- Repeat the test for the target language screen

## Screenshots

Loading state | Error state
----- | ----- 
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/97739cd8-cf94-4029-bde9-f436a5dce3ec" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/74055710-5fa7-44ac-a68a-3a442a9379e8" width=320 >



---
